worlds/lingo/static_logic.py

```diff
@@ -0,0 +1,512 @@
+    name: str
+    entrances: List[RoomEntrance]
+
+
+class Door(NamedTuple):
+    name: str
+    item_name: str
+    location_name: Optional[str]
+    panels: Optional[List[RoomAndPanel]]
+    skip_location: bool
+    skip_item: bool
+    door_ids: List[str]
+    painting_ids: List[str]
+    event: bool
+    group: Optional[str]
+    include_reduce: bool
+    junk_item: bool
+
+
+class Panel(NamedTuple):
+    required_rooms: List[str]
+    required_doors: List[RoomAndDoor]
+    required_panels: List[RoomAndPanel]
+    colors: List[str]
+    check: bool
+    event: bool
+    internal_ids: List[str]
+    exclude_reduce: bool
+    achievement: bool
+    non_counting: bool
+
+
+class Painting(NamedTuple):
+    id: str
+    room: str
+    enter_only: bool
+    exit_only: bool
+    orientation: str
+    required: bool
+    required_when_no_doors: bool
+    required_door: Optional[RoomAndDoor]
+    disable: bool
+    move: bool
+
+
+class Progression(NamedTuple):
+    item_name: str
+    index: int
+
+
+class StaticLingoLogic:
+    ROOMS: Dict[str, Room]
+    PANELS: Dict[str, Panel]
+    DOORS: Dict[str, Door]
+    PAINTINGS: Dict[str, Painting]
+
+    ALL_ROOMS: List[Room]
+    DOORS_BY_ROOM: Dict[str, Dict[str, Door]]
+    PANELS_BY_ROOM: Dict[str, Dict[str, Panel]]
+    PAINTINGS_BY_ROOM: Dict[str, List[Painting]]
+
+    PROGRESSIVE_ITEMS: List[str]
+    PROGRESSION_BY_ROOM: Dict[str, Dict[str, Progression]]
+
+    PAINTING_ENTRANCES: int
+    PAINTING_EXIT_ROOMS: Set[str]
+    PAINTING_EXITS: int
+    REQUIRED_PAINTING_ROOMS: List[str]
+    REQUIRED_PAINTING_WHEN_NO_DOORS_ROOMS: List[str]
+
+    SPECIAL_ITEM_IDS: Dict[str, int]
+    PANEL_LOCATION_IDS: Dict[str, Dict[str, int]]
+    DOOR_LOCATION_IDS: Dict[str, Dict[str, int]]
+    DOOR_ITEM_IDS: Dict[str, Dict[str, int]]
+    DOOR_GROUP_ITEM_IDS: Dict[str, int]
+    PROGRESSIVE_ITEM_IDS: Dict[str, int]
+
+    def __init__(self):
+        self.ROOMS = {}
+        self.PANELS = {}
+        self.DOORS = {}
+        self.PAINTINGS = {}
+
+        self.ALL_ROOMS = []
+        self.DOORS_BY_ROOM = {}
+        self.PANELS_BY_ROOM = {}
+        self.PAINTINGS_BY_ROOM = {}
+
+        self.PROGRESSIVE_ITEMS = []
+        self.PROGRESSION_BY_ROOM = {}
+
+        self.PAINTING_ENTRANCES = 0
+        self.PAINTING_EXIT_ROOMS = set()
+        self.PAINTING_EXITS = 0
+        self.REQUIRED_PAINTING_ROOMS = []
+        self.REQUIRED_PAINTING_WHEN_NO_DOORS_ROOMS = []
+
+        self.SPECIAL_ITEM_IDS = {}
+        self.PANEL_LOCATION_IDS = {}
+        self.DOOR_LOCATION_IDS = {}
+        self.DOOR_ITEM_IDS = {}
+        self.DOOR_GROUP_ITEM_IDS = {}
+        self.PROGRESSIVE_ITEM_IDS = {}
+
+        try:
+            from importlib.resources import files
+        except ImportError:
+            from importlib_resources import files
+
+        with files("worlds.lingo").joinpath("ids.yaml").open() as file:
+            config = yaml.load(file, Loader=yaml.Loader)
+
+            if "special_items" in config:
```

Comment:
Since config is a static file don't you already know what's in it? I also think it'd be better to throw an error rather than just silently ignore missing data?

Reply:
Errors do already get thrown in the get_*_id functions if there is missing data.

Reply:
In that case all these if checks do is make this slower.